src/control/server/storage/bdev.go

```diff
@@ -31,35 +30,36 @@ import (
 // BdevPciAddrSep defines the separator used between PCI addresses in string lists.
 const BdevPciAddrSep = " "
 
-// BioState represents the health state of BIO device on an NVMe device (a blobstore on a NVMe
-// namespace).
-type BioState uint32
+// NvmeDevState represents the health state of NVMe device as reported by DAOS engine BIO module.
+type NvmeDevState uint32
 
-// BioState constant definitions to represent expected bitset flag combinations.
+// NvmeDevState constant definitions to represent expected bitset flag combinations.
 const (
-	BioStateNew      BioState = C.NVME_DEV_FL_PLUGGED
-	BioStateNormal   BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE
-	BioStateFaulty   BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE | C.NVME_DEV_FL_FAULTY
-	BioStateIdentify BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE | C.NVME_DEV_FL_IDENTIFY
+	NvmeDevStateNew      NvmeDevState = C.NVME_DEV_FL_PLUGGED
+	NvmeDevStateNormal   NvmeDevState = NvmeDevStateNew | C.NVME_DEV_FL_INUSE
+	NvmeDevStateFaulty   NvmeDevState = NvmeDevStateNormal | C.NVME_DEV_FL_FAULTY
+	NvmeDevStateIdentify NvmeDevState = NvmeDevStateNormal | C.NVME_DEV_FL_IDENTIFY
```

Comment:
Does there need to be a state to indicate that a device is no longer plugged in but is still in the list of devices? For example, when a device is unplugged, what does that bitmask look like?

Reply:
Unplugged device will have bitmask 0 (bio will handle resetting of other flags) but I don't think we need to have a control-plane state constant to represent the specific state as we don't handle it other than to print the StatusString or set of flags (Stringer). In the case that a SMD entry exists in an unplugged state, we will still be able to query the SMD entry and the state will remain unplugged until the application restarts (failed attempts to retrieve health for the entry will be ignored during query).